fastly/block_fastly_service_v1_package.go

```diff
@@ -0,0 +1,107 @@
+package fastly
+
+import (
+	"fmt"
+	"log"
+
+	gofastly "github.com/fastly/go-fastly/fastly"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+type PackageServiceAttributeHandler struct {
+	*DefaultServiceAttributeHandler
+}
+
+func NewServicePackage() ServiceAttributeDefinition {
+	return &PackageServiceAttributeHandler{
+		&DefaultServiceAttributeHandler{
+			key: "package",
+		},
+	}
+}
+
+func (h *PackageServiceAttributeHandler) Register(s *schema.Resource) error {
+	s.Schema[h.GetKey()] = &schema.Schema{
+		Type:     schema.TypeList,
+		Required: true,
+		MaxItems: 1,
+		MinItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"filename": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+				// sha512 hash of the file
+				"source_code_hash": {
+					Type:     schema.TypeString,
+					Optional: true,
+					Computed: true,
+				},
+				"source_code_size": {
+					Type:     schema.TypeInt,
+					Computed: true,
+				},
+			},
+		},
+	}
+	return nil
+}
+
+func (h *PackageServiceAttributeHandler) Process(d *schema.ResourceData, latestVersion int, conn *gofastly.Client) error {
+
+	if v, ok := d.GetOk(h.GetKey()); ok {
+		// Schema guarantees one package block.
+		Package := v.([]interface{})[0].(map[string]interface{})
+		packageFilename := Package["filename"].(string)
```

Comment:
Does this panic if you have an empty `package` resource block since the `filename` field is optional and doesn't set an explicit default of `""`? Is there an implicit default of `""` for `schema.TypeString`?

Reply:
Filename is now required